src/lib/containers/StatisticsPlot.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react'
+// tslint:disable-next-line:import-name
+import Plotly from 'plotly.js-basic-dist'
+// tslint:disable-next-line:import-name
+import createPlotlyComponent from 'react-plotly.js/factory'
+import { SynapseClient } from '../utils'
+import { ProjectFilesStatisticsRequest, ProjectFilesStatisticsResponse, FilesCountStatistics } from '../utils/jsonResponses/Statistics'
+const Plot = createPlotlyComponent(Plotly)
+
+export type StatisticsPlotProps = {
+  request: ProjectFilesStatisticsRequest
+  token?: string
+  endpoint?: string
+  title?: string
+  xtitle?: string
+  ytitle?: string
+  isHorizontal?: boolean
+  xaxistype?: string
+  showlegend?: boolean
+}
+
+type StatisticsPlotState = {
+  isLoaded: boolean
+  plotData?: ProjectFilesStatisticsResponse
+}
+
+class StatisticsPlot extends React.Component<StatisticsPlotProps, StatisticsPlotState> {
+
+  constructor(props: StatisticsPlotProps) {
+    super(props)
+    this.state = {
+      isLoaded: false,
+      plotData: {} as ProjectFilesStatisticsResponse
+    }
+    this.fetchPlotlyData = this.fetchPlotlyData.bind(this)
```

Comment:
I think you can avoid having to do bind those functions by having them as class props e.f.
`fetchPlotlyData = () =>  {
...` instead of `  public fetchPlotlyData() { ...`